ssl/ssl_lib.c

```diff
@@ -431,6 +431,35 @@ static int dane_tlsa_add(SSL_DANE *dane,
     return 1;
 }
 
+/*
+ * Return 0 if there is only one version configured and it was disabled
+ * at configure time.  Return 1 otherwise.
+ */
+static int ssl_check_allowed_versions(int min_version, int max_version)
+{
+    if (min_version == max_version
```

Comment:
I'm not really convinced it needs to be that complicated. Check if max_version is 0 at the beginning and if so set it to TLS1_3_VERSION. Check if min_version is 0 at the beginning and if so set it to SSL3_VERSION. Then a series of checks like this:

````
#ifdef OPENSSL_NO_TLS1_3
    if (max_version == TLS1_3_VERSION)
        max_version = TLS1_2_VERSION;
#endif
#ifdef OPENSSL_NO_TLS1_2
    if (max_version == TLS1_2_VERSION)
        max_version = TLS1_1_VERSION;
#endif
...and so on...
#ifdef OPENSSL_NO_SSL3
    if (max_version == SSL3_VERSION)
        max_version = 0;
#endif
    if (max_version < min_version)
        return 0;
    return 1;
````